src/main/java/org/apache/sysml/runtime/matrix/data/LibMatrixCUDA.java

```diff
@@ -300,57 +327,416 @@ public static void matmultTSMM(MatrixObject left, MatrixObject output,
 	    if(!output.getGPUObject().isAllocated)
 	            throw new DMLRuntimeException("Output is not allocated:" + output.getGPUObject().isAllocated);
 	
-	    Pointer A = ((JCudaObject)left.getGPUObject()).jcudaPointer;
-	    Pointer C = ((JCudaObject)output.getGPUObject()).jcudaPointer;
+	    Pointer A = ((JCudaObject)left.getGPUObject()).jcudaDenseMatrixPtr;
```

Comment:
We need to do sparse to dense conversion here if the GPU object is in sparse format.